ouroboros-consensus-test/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -783,8 +800,12 @@ withTestMempool setup@TestSetup {..} prop =
       -- Set up the LedgerInterface
       varCurrentLedgerState <- uncheckedNewTVarM testLedgerState
       let ledgerInterface = LedgerInterface
-            { getCurrentLedgerState = readTVar varCurrentLedgerState
-            , getLedgerTablesAtFor = \_ _ -> pure $ Right polyEmptyLedgerTables
+            { getCurrentLedgerState = forgetLedgerTables <$> readTVar varCurrentLedgerState
+            , getLedgerTablesAtFor = \pt _ -> do
+                st <- atomically $ readTVar varCurrentLedgerState
+                if castPoint (getTip st) == pt
+                  then pure $ Right $ projectLedgerTables st
+                  else pure $ Left $ PointNotFound pt
```

Comment:
Good point. I don't know really. One argument for not considering cases other than the tip is that we are not exercising this at the moment, but maybe we should. 

The advantage of using the backing store is that this would allow us to test the integration with an actual backing store, but this will complicate the tests, since we will have to generate actions that mock every possible alteration in the backing store. So I'd be inclined to test the ledger interface and mempool separately. 

So maybe we could keep a list of past ledger states we have seen previously and only support those points.